lib/vrl/stdlib/src/format_timestamp.rs

```diff
@@ -4,6 +4,14 @@ use chrono::{
 };
 use vrl::prelude::*;
 
+fn format_timestamp(bytes: Value, ts: Value) -> Resolved {
+    let bytes = bytes.try_bytes()?;
```

Comment:
This is a general theme that I have noticed where I see some optimization potential - especially compared to the traverse model. When taking into account the `compiler::State` information, the VM should know when it can safely unwrap this type or must check it at runtime. Then the concrete function implementation doesn't need to check it again.

That definitely warrants another PR, though.